### DISCUSSION TITLE
read image width within js

### DISCUSSION THREAD

**Discussion by s-light** [quasarframework/quasar#13571] [General - Components / Directives / etc] [answered]
hello community,

i try to do some image upload pre-processing..
and in this i need a possibility to get the image-width and height.
i have tried with something like this:
```
                const img = new Image()
                console.log("img", img);
                img.scr = URL.createObjectURL(file)
                console.log("img.scr", img.scr);
                img.decode()
                .then(() => {
                    console.log("decode done....");
                    const sizes = {
                        width:this.width,
                        height: this.height
                    };
                    URL.revokeObjectURL(img.src);
                    console.log("sizes", sizes);
                })
                .catch((encodingError) => {
                    console.log("error", encodingError);
                })
```
and getting every time `error DOMException: Invalid image request.`
so it seems i do something wrong.

my stripped down example of this is at
https://jsfiddle.net/s_light/mc2kofwn/

does anybody has any tips or hints how i can do it?

sunny greetings
stefan

**Comment by MilosPaunovic** [comment]
Google has hints:
[https://stackoverflow.com/questions/8903854/check-image-width-and-height-before-upload-with-javascript](https://stackoverflow.com/questions/8903854/check-image-width-and-height-before-upload-with-javascript
)
This is not Quasar related.

**Reply by s-light** [reply]
the hint is more or less exactly what i have done..
i had one typo (see accepted answer)
and as i did not find what could cause this i thought it had to do something with my used framework / code around...

**Comment by pdanpdan** [comment, marked as answer]
The error is correct, you don't set a `src` attribute on the img so it doesn't know what to decode.

**Reply by s-light** [reply]
hmpf.....
you are correct
i have a typo in my code...
`img.scr = URL.createObjectURL(file)`
should be
`img.src = URL.createObjectURL(file)`

thanks for the *heads up*

**Reply by pdanpdan** [reply]
:)
